lib/grppclat.gi

```diff
@@ -432,15 +432,32 @@ local P,g,op,act,a,pcgs,ma,mat,d,f,i,j,new,newmat,id,p,dodim,compldim,compl,
   ma:=new;
 
   # convert to grps
+  pccache:=[]; # avoid recerating different copies of same element 
+  pcelm:=function(vec)
+  local e,p;
+    e:=Immutable([vec]);
+    p:=PositionSorted(pccache,e);
+    if IsBound(pccache[p]) and pccache[p][1]=vec then
+      return pccache[p][2];
+    else
+      e:=Immutable([vec,PcElementByExponentsNC(pcgs,vec)]);
+      AddSet(pccache,e);
+      return e[2];
+    fi;
+  end;
+
   new:=[];
   for i in ma do
     #a:=SubgroupNC(Parent(g),List(i,j->Product([1..d],k->pcgs[k]^j[k])));
+    asz:=kersz*p^Length(i);
     if kersz=1 then
-      a:=SubgroupNC(par,List(i,j->PcElementByExponentsNC(pcgs,j)));
+      a:=SubgroupNC(par,List(i,j->pcelm(j)));
     else
-      a:=ClosureGroup(ker,List(i,j->PcElementByExponentsNC(pcgs,j)));
+      #a:=ClosureGroup(ker,List(i,j->pcelm(j)):knownClosureSize:=asz);
```

Comment:
Why did you replace the `ClosureGroup` with the `SubgroupNC`? Is it faster? And given the replacement, why have you kept the `ClosureGroup` line as a comment?